src/rgw/rgw_public_access.h

```diff
@@ -0,0 +1,70 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 SUSE LLC
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#pragma once
+#include <include/types.h>
+
+class XMLObj;
+
+namespace rgw::IAM {
+
+class PublicAccessConfiguration {
```

Comment:
is PublicAccessConfiguration different than the PublicAccessBlockConfiguration described in s3 docs? https://docs.aws.amazon.com/AmazonS3/latest/API/API_PublicAccessBlockConfiguration.html

if not, can we rename this to PublicAccessBlockConfiguration and move it out of the IAM namespace?

Reply:
it is the same thing, I'll change.